Add Mantine theme

zfarhad opened the pull request:
### Reasons for making this change

Added a new theme using [Mantine UI](https://mantine.dev/) library (@mantine/core, @mantine/hooks, @mantine/dates) for react-jsonschema-form, addressing the request in issue #4289 (fixes #4289). This theme provides a modern, customizable UI component set that integrates well with RJSF's form generation capabilities.

The Mantine theme offers:
- Modern, clean UI components
- Built-in dark mode support
- Responsive design
- Accessible components
- Customizable styling through Mantine's theming system

### Checklist

- [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://rjsf-team.github.io/react-jsonschema-form/docs/contributing) of the Markdown text I've added
- [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests. I've run `npx nx run-many --target=build --exclude=@rjsf/docs && npm run test:update` to update snapshots, if needed.
  - [x] I've updated [docs](https://rjsf-team.github.io/react-jsonschema-form/docs) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/main/CHANGELOG.md) with a description of the PR
- [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

heath-freenome commented:
> @zfarhad No new features will be accepted to version 5. Please make a PR for [version 6](https://github.com/rjsf-team/react-jsonschema-form/tree/rjsf-v6) instead, which currently is in alpha.
> 
> _(I'm working on the primereact theme #4485 and was asked to do the same)_

@zfarhad Yes, @jasny is correct. We are only accepting critical fixes to the v5 release. Please make this change into the `rjsf-v6` branch instead. Also, you will need to make changes to the `ArrayFieldItemTemplate` and add a `GridTemplate`. See the `migrate v6 migration guide` for more details.

heath-freenome commented:
Also, feel free to join the Friday meeting mentioned in the pinned issue

zfarhad commented:
> > @zfarhad No new features will be accepted to version 5. Please make a PR for [version 6](https://github.com/rjsf-team/react-jsonschema-form/tree/rjsf-v6) instead, which currently is in alpha.
> > _(I'm working on the primereact theme #4485 and was asked to do the same)_
> 
> @zfarhad Yes, @jasny is correct. We are only accepting critical fixes to the v5 release. Please make this change into the `rjsf-v6` branch instead. Also, you will need to make changes to the `ArrayFieldItemTemplate` and add a `GridTemplate`. See the `migrate v6 migration guide` for more details.

Sounds great, I didn't know you were developing v6. So I will prepare the theme based on the rjsf-v6 branch and submit a new PR request on it.

JoeHillyard commented:
Is there anything I can do to help get this over the line?

heath-freenome commented:
@zfarhad Have you had a chance to make your changes against the `rjsf-v6` branch?

heath-freenome commented:
@zfarhad I've merged a few PRs with updates and fixes from `main`. You'll have to rebase and update your PR. Also, checkout the new `Layout Grid` in the playground and update the `layoutGrid.tsx` in the `samples` directory to add your custom `uiSchema` for the grid to it.

zfarhad commented:
> @zfarhad I've merged a few PRs with updates and fixes from `main`. You'll have to rebase and update your PR. Also, checkout the new `Layout Grid` in the playground and update the `layoutGrid.tsx` in the `samples` directory to add your custom `uiSchema` for the grid to it.

@heath-freenome Sorry for my late, I will migrate the theme to the rjsf-v6 branch later this week

dashkan commented:
@zfarhad Great job!

I don't know if this has been addressed in v6 layout. All the submit buttons for mantine theme are off in playground.

**default theme:**

<img width="714" alt="image" src="https://github.com/user-attachments/assets/e930135c-8c2c-4cd3-8508-d36ddfa3f4ed" />

<img width="714" alt="image" src="https://github.com/user-attachments/assets/51197b11-5d97-4a62-8639-faebf332be29" />

**mantine theme:**
<img width="714" alt="image" src="https://github.com/user-attachments/assets/b6ad57c4-62b2-466b-80aa-58fc596d89f0" />

custom example in playground does not look right either
<img width="714" alt="image" src="https://github.com/user-attachments/assets/175b66df-c4cf-43a2-bf56-15ccfd66c38c" />

heath-freenome commented:
@zfarhad still looking forward to seeing this rebased to the `rjsf-v6` branch

heath-freenome commented:
@zfarhad Just a reminder that we are not accepting any new features into the main build. Can you rebase your PR onto the `rjsf-v6` branch?

heath-freenome commented:
> @heath-freenome I've rebased the theme on the rjsf-v6 branch, it's ready now. Do I need to open a new PR for merging or will you continue with this one?

Can you update this PR to merge into the `rjsf-v6` branch instead of `main`? If not, then you'll need a new PR.

zfarhad commented:
> > @heath-freenome I've rebased the theme on the rjsf-v6 branch, it's ready now. Do I need to open a new PR for merging or will you continue with this one?
> 
> Can you update this PR to merge into the `rjsf-v6` branch instead of `main`? If not, then you'll need a new PR.

I changed the base branch to [rjsf-v6](https://github.com/rjsf-team/react-jsonschema-form/tree/rjsf-v6) but I couldn't change the head branch of the pull request, So I have created a [new PR](https://github.com/rjsf-team/react-jsonschema-form/pull/4560) to merge into rjsf-v6.

heath-freenome commented:
@zfarhad can we close this?

zfarhad commented:
> @zfarhad can we close this?

Done
